library/common/extensions/filter/platform_filter.cc

```diff
@@ -0,0 +1,71 @@
+#include "library/common/extensions/filter/platform_filter.h"
+
+#include "common/common/assert.h"
+#include "common/common/utility.h"
+
+#include "library/common/buffer/utility.h"
+#include "library/common/http/header_utility.h"
+
+namespace Envoy {
+namespace Http {
+
+FilterHeadersStatus mapStatus(envoy_filter_headers_status_t status) {
+  switch (status) {
+  case ENVOY_FILTER_HEADERS_STATUS_CONTINUE:
+    return FilterHeadersStatus::Continue;
+  case ENVOY_FILTER_HEADERS_STATUS_STOP_ITERATION:
+    return FilterHeadersStatus::StopIteration;
+  case ENVOY_FILTER_HEADERS_STATUS_CONTINUE_AND_END_STREAM:
+    return FilterHeadersStatus::ContinueAndEndStream;
+  case ENVOY_FILTER_HEADERS_STATUS_STOP_ALL_ITERATION_AND_BUFFER:
+    return FilterHeadersStatus::StopAllIterationAndBuffer;
+  default:
+    ASSERT(false, "unrecognized filter status from platform: {}");
+    return FilterHeadersStatus::Continue;
+  }
+}
+
+FilterHeadersStatus PlatformExtensionFilter::decodeHeaders(RequestHeaderMap& headers,
+                                                           bool end_stream) {
+  return mapStatus(platform_filter_.on_request_headers(Utility::toBridgeHeaders(headers),
```

Comment:
`Utility::toBridgeHeaders` copies the headers. If the platform modifies the values then it would not be reflected in the `RequestHeaderMap& headers`. We can solve this a few ways (as of course you know). I might bias toward on_request_headers returning the value? It would cause a second copy but allows us to keep the memory contract were the caller of the function owns memory cleanup, i.e the platform code would own cleaning the passed in value, and this code would on clean up of the return value after we update `RequestHeaderMap& headers` here.

wdyt?

Reply:
discussed offline, will add more context to the PR/comments